common/hyperscaler/rules/rules_service.go

```diff
@@ -92,6 +92,20 @@ func (rs *RulesService) parse(rulesConfig *RulesConfig) *ParsingResults {
 	return results
 }
 
+// Match finds the matching rule for the given provisioning attributes and provide the set of labels, which must be used to find proper secret binding.
+func (rs *RulesService) MatchProvisioningAttributes(provisioningAttributes *ProvisioningAttributes) (Result, bool) {
+	var result Result
+	found := false
+	for _, parsingResult := range rs.Parsed.Results {
+		if !parsingResult.HasParsingErrors() && parsingResult.Rule.Matched(provisioningAttributes) {
```

Comment:
Does it mean that we check parsing errors each time we try to calculate labels?
If there are parsing errors we shouldn't try to match. 
In case of KEB we should fail to start.
In case of CLI command we should fail as well.
If KEB is running we should assume there is no parsing errors. Unless we are very, very defensive.

To sum up.
In KEB we parse ruleset once. If it is not ok, we abort.
If it is ok, we can match provisioning attributes for each instance creation assuming that rules are valid, unambiguous, unique.